### ISSUE TITLE
Provide Gemini mime type icon

### ISSUE THREAD

**Issue by Antolius** [starfish-app/Starfish#15]
Currently installing Starfish registers Gemini mime type on the system. It should provide a custom icon for it as well.

**Comment by Suzie97** [comment]
@Antolius the icon is going to be just an svg right? Let me know the sizes needed, I can make some

**Comment by Suzie97** [comment]
> For design I was thinking of making it distinct from app icon. That way users can have multiple clients installed (e.g. Starfish & Lagrange) and even use some other app as the default without file icons looking out of place. We could use a zodiac inspired ( ♊ ) sign in the icon. Or perhaps explore [NASA's Gemini missions]

So the icon will only be visible if users install Starfish? What if someone does not have Starfish installed and uses something else like Lagrange instead? I think if we want to make a generic icon, we should make a PR to the [elementary icons](https://github.com/elementary/icons) repo right?

**Comment by Antolius** [comment]
> So the icon will only be visible if users install Starfish?

Yeah, currently registering the mime type happens at the time of Starfish install. I'm not sure is other Gemini clients do the same. Generally having a local file-type defined is not really necessary, as the client loads content from the internet in most cases. A local file-type is useful only for users that have  [gemtext markup files](https://gemini.circumlunar.space/docs/gemtext.gmi) stored locally on their PCs. One use-case would be Gemini authors who write markup files directly. In this case a client such as Starfish, which is capable of displaying a `file:///path/to/local/file.gmi` URI, can be used to preview those gemtext files locally without having to upload them to the server first.

With that in mind, adding the `text/gemini` mime type on elementary OS level would be cool, but I'm not sure if maintainers would be interested in it. Also, users will probably be able to install newer versions of Starfish on older elementary OS versions, meaning that even if OS starts shipping the gemini mime type (definition and icon) at some point I think there's also value in making sure it exists at the moment of Starfish install (or update). So I guess the best approach would be to add icon to the app and then offer to port it upstream (to the OS level). We'll need to check how overrides are handled though (i.e. what happens if both OS and the app try to define the same mime type).

**Comment by Antolius** [comment]
I looked into this one a bit more, and `text/gemini` mime type is not yet officially registered, as can(not) be seen in the [IANA media types list](https://www.iana.org/assignments/media-types/media-types.xhtml). There's been some discussion about this in, as far as I know unofficial, [GitLab issue tracker for the Gemini protocol](https://gitlab.com/gemini-specification/gemini-text/-/issues/11). Gemini mailing list is currently down so I can't check there.

As for other browsers installing the mime type for `.gmi` file extension, I see that [Kristal browser](https://kristall.random-projects.net/) does it [here](https://github.com/MasterQ32/kristall/blob/master/kristall-mime-info.xml#L3-L8) (and it's a bit more detailed than what Starfish does, so we could probably learn from it). But as far as I can see it doesn't provide an icon for it. I looked for it in Lagrange's code, but as far as I understand it [only does it for iOS and MacOS](https://github.com/skyjake/lagrange/search?l=XML&q=gmi&type=code) (though I might be reading this config wrong, I'm not really familiar with it).

At this point I think it's better to define the mime type (and its icon) on the application level, until it's been officially registered. I also think it would be a good idea to go with a more inclusive icon, so that we can pull it upstream to elementary level at some point in the future.

**Comment by Suzie97** [comment]
> At this point I think it's better to define the mime type (and its icon) on the application level, until it's been officially registered. I also think it would be a good idea to go with a more inclusive icon, so that we can pull it upstream to elementary level at some point in the future. 

Great, my exam was over today, so now I can get back to work. I'll share a few sketches here

**Comment by Suzie97** [comment]
> But as far as I can see it doesn't provide an icon for it. I looked for it in Lagrange's code, but as far as I understand it [only does it for iOS and MacOS](https://github.com/skyjake/lagrange/search?l=XML&q=gmi&type=code) (though I might be reading this config wrong, I'm not really familiar with it).

@Antolius I think Lagrange does not provide an icon both on iOS and macOS. I tested it on my machines and there are no icons

## macOS
<img width="131" alt="Screenshot 2022-01-23 at 21 47 51" src="https://user-images.githubusercontent.com/68198116/150687740-b6cd95e7-d1ba-40f8-b027-7386ce9dc74a.png">

## iOS
![IMG_E7AA453C5546-1](https://user-images.githubusercontent.com/68198116/150687755-2752b2ef-0e76-43e7-ade8-787727ac5ac1.jpeg)
